src/fcc_bridge_node_ros.cpp

```diff
@@ -0,0 +1,225 @@
+//
+// Created by Johan <job8197@thi.de> on 30.04.2024.
+//
+
+#include "fcc_bridge_node.hpp"
+
+// Libc header
+#include <chrono>
+
+namespace fcc_bridge {
+FCCBridgeNode::FCCBridgeNode(const std::string &name/*,
+                             const rclcpp::NodeOptions &node_options*/)
+    : CommonNode(name), internal_state(INTERNAL_STATE::STARTING_UP) {
+    // Pre-populate the last mission control heartbeat
+    this->last_mission_control_heartbeat.time_stamp = this->now();
+
+    // Setup ROS objects such as timer, publishers etc.
+    this->setup_ros();
+    if (this->internal_state == INTERNAL_STATE::ERROR) {
+        RCLCPP_FATAL(this->get_logger(), "Failed to setup ROS! Exiting...");
+        this->exit_process_on_error();
+    }
+    this->internal_state = INTERNAL_STATE::ROS_SET_UP;
+    RCLCPP_INFO(this->get_logger(), "Transitioning into ROS_SET_UP state");
+
+    // Setup MAVSDk objects such as system, telemetry etc.
+    this->setup_mavsdk();
+    if (this->internal_state == INTERNAL_STATE::ERROR) {

```

Comment:
I would use a getter instead of accessing the value directly

Reply:
The member variable is already private so adding a getter does not make things better guarded.

Reply:
This way you can add Log msgs or more functionality into these getters later on, that you might not now yet.

Reply:
Implemented in b831b9d